### PR TITLE
Eliminate vm.heapBaseAddress()

### PR DESCRIPTION
**Pull request by dchopra001**
This commit removes `OMR::VMEnv::heapBaseAddress()` as this routine always returns 0. It was introduced in the codebase early on in the codebase and has since been superseded by a shift-based solution. Hence this is no longer needed. See https://github.com/eclipse/openj9/issues/11263 for original issue and discussion.

Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

### CODE REVIEW

**Comment by Leonardo2718** [comment]
What's the reason for doing this? The commit message doesn't have a description.

**Comment by dchopra001** [comment]
> What's the reason for doing this? The commit message doesn't have a description.

Sorry about that I've updated it now.

File: compiler/optimizer/GlobalRegisterAllocator.cpp [hunk 1]
```diff
@@ -350,20 +350,6 @@ TR_GlobalRegisterAllocator::perform()
...
 
-   if (comp()->useCompressedPointers() &&
-       comp()->cg()->materializesHeapBase() &&
-       TR::Compiler->vm.heapBaseAddress() != 0)
-      {
-      TR::SymbolReference *autoSymRef = NULL;
-      vcount_t visitCount = comp()->incVisitCount();
-      for (TR::TreeTop * tt = comp()->getStartTree(); tt; tt = tt->getNextTreeTop())
-         {
-         changeHeapBaseConstToLoad(comp(), autoSymRef, tt->getNode(), visitCount);
```
Comment (fjeremic) [inline compiler/optimizer/GlobalRegisterAllocator.cpp:361]: As far as I can see if you eliminate the call to `changeHeapBaseConstToLoad` here then this function is dead. Can we remove it?

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -5449,34 +5447,6 @@ istoreHelper(TR::Node * node, TR::CodeGenerator * cg, bool isReversed)
             translatedNode = translatedNode->getFirstChild();
          if (translatedNode->getOpCode().isRightShift()) // optional
             translatedNode = translatedNode->getFirstChild();
```
Comment (fjeremic) [inline compiler/z/codegen/OMRTreeEvaluator.cpp:5449]: This code involving `translatedNode` seems to have no use. Can we remove it? Similarly in the other codegens.

**Review by fjeremic — CHANGES_REQUESTED** [review]

**Review by fjeremic — APPROVED** [review]

**Comment by fjeremic** [comment]
@genie-omr build all

**Pull request merged**